Rango for Safari

nriley opened the pull request:
Wrap Rango as a Safari web extension. Implements clipboard support in the native app.

Update CONTRIBUTING.md with Safari instructions.

Here's the app's main window:

<img width="537" alt="image" src="https://user-images.githubusercontent.com/68303/178125210-8f310ac5-7c46-4be1-917e-0289a4ef6fdb.png">

nriley committed:
Read text from clipboard in Safari.

Sets the get-talon-request key to Ctrl+Shift+1 as Ctrl+Shift+Insert doesn't work.

nriley committed:
Remove debug logging from web extension background page.

nriley committed:
Correct the extension bundle identifier so the app will appropriate identify whether Rango is active.

nriley committed:
Don't launch the app in the debugger (speeds up testing substantially).

nriley committed:
Write to the clipboard using a text area.

From https://github.com/david-tejada/rango/compare/main...safari-clipboard#diff-9d5b53d5d2c8c5ce5f718e9cfc117d98008818b6071ce1902d74956ddf2b8f8cR45.

nriley committed:
Remove iOS specific implementations in ViewController.

nriley committed:
Declare deployment target in fewer places.

nriley committed:
Remove iOS icons.

nriley committed:
Make Help menu item open Rango's GitHub page.

nriley committed:
Update deployment target as there are some issues rendering the main window on 10.15.

(However, it did work fine on 10.15.)

nriley committed:
Add a version and enable apple-generic versioning.

agvtool doesn't understand generated Info.plist files, so update the marketing version in Xcode.

nriley committed:
Code sign, though this doesn't count for Safari any more.

nriley committed:
Include a note about having to turn on Allow Unsigned Extensions.

nriley committed:
Chrome doesn't define navigator.vendor.

nriley committed:
Change shortcut to Ctrl+Shift+1 on macOS in browsers using manifest version 3.

nriley committed:
Merge branch 'main' into safari.

Update Rango app to match release version 0.2.

nriley committed:
Merge branch 'main' into safari

nriley committed:
Rango for Safari 0.2.1.

Also update deployment target as Xcode no longer allows me to use 12.4.

nriley committed:
Merge branch 'main' into safari

nriley committed:
Move MARKETING_VERSION into generated xcconfig file.

nriley committed:
Update CONTRIBUTING.md with Safari instructions.

david-tejada commented:
All seems okay (I guess 😅). I leave you the logos at different sizes in case you want to use those. If you think the ones you have are okay let me know and I can merge this one
[Rango logo.zip](https://github.com/david-tejada/rango/files/9102698/Rango.logo.zip)
.

david-tejada commented:
I did some file reorganization and renaming. Now clipboard.ts is in `src/background/utils/clipboard.ts`. What's the best way to solve the conflict, you do a rebase?

david-tejada commented:
I was trying to resolve the merge conflicts myself and I realized that you have to request "nativeMessaging" permission in Safari. I think we are going to need to create a different manifest file for Safari, because requesting permissions that you're not using could be problematic

nriley commented:
> I was trying to resolve the merge conflicts myself and I realized that you have to request "nativeMessaging" permission in Safari. I think we are going to need to create a different manifest file for Safari, because requesting permissions that you're not using could be problematic

OK, whatever works, just let me know if you want me to do anything. Should have some time this weekend again.

david-tejada commented:
I'll create another manifest and commands for building to dist-safari. I just need to know if there's anything else that needs to be changed from your code to reflect this change

nriley commented:
> I'll create another manifest and commands for building to dist-safari. I just need to know if there's anything else that needs to be changed from your code to reflect this change 

Would just need to change the Xcode project file to point at the new manifest (and build folder if appropriate). Should be a 5 min change once you have the new manifest/build command done.

nriley commented:
FYI, I am holding off on revising this PR until you let me know you're ready!

david-tejada commented:
Yes, sorry. I'll do it this week so you can do your part on the weekend

david-tejada commented:
I created the separate manifest and build scripts for Safari.

Btw, I bought an air m1 because the Linux machine I was using was very slow and was getting on my nerves. So next time I can try to package it myself for Safari

nriley committed:
Merge branch 'main' into safari

nriley committed:
Switch to dist-mv2-safari.

nriley committed:
Undo Safari-specific manifest changes.

nriley committed:
Use macOS 11+ logging with a subsystem.

Stream logs with:

/usr/bin/log stream --predicate 'subsystem == "net.sabi.Rango.extension"' --info

nriley commented:
I've updated (finally) to the new manifest and undid my changes from the old one. May want to add a debugging guide, but here's some things to be aware of:

1) Don't ever have more than one copy of Rango.app on your system or results may be unpredictable (use Product > Clean Build Folder in Xcode if needed).
2) Some things that can help if Rango doesn't appear in Safari: uncheck/check in Safari Extensions Preferences; restart Safari.
3) Messages logged by the extension do not appear in Xcode but are logged to the system log (stream with `/usr/bin/log stream --predicate 'subsystem == "net.sabi.Rango.extension"' --info`).

david-tejada commented on CONTRIBUTING.md:
````diff
@@ -42,3 +42,30 @@ _This example is for launching Brave on MacOS_
+1. Build the extension for manifest version 2:
+
+    ```bash
+    npm run build:mv2
````
I think this should be `npm run build:mv2-safari`.

nriley replied:
Oops, fixed.

nriley committed:
Update npm instructions to use the Safari-specific manifest.
